### PR TITLE
Legacy Code (BE) for testing userGenders

### LINKED ISSUE #105: Add test for userGender
Add happy and sad path testing for userGender in user QUERY

### PR DESCRIPTION
**Pull request by priyapower**
**What was changed**
- Adds testing (happy and sad paths) for the QUERY userGenders

closes #105

### CODE REVIEW

File: code/api/src/modules/user/tests/userMutation.test.js [hunk 1]
```diff
@@ -47,49 +47,68 @@ describe('User mutations', () => {
...
         query:
-          `mutation { userUpdate(id: ${id}, image: "updated_img.png", description: "new day, new me", address_line1: "NEW ADDRESS LINE 1", address_line2: "NEW ADDRESS LINE 2", city: "NEW CITY", state: "AA", zipcode: 11111 ) { id image description address_line1 address_line2 city state zipcode } }`
+          `mutation { userUpdate(id: ${id}, name: "NEW NAME", email: "EMAIL@EMAIL.COM", password: "123456", image: "updated_img.png", description: "new day, new me", address_line1: "NEW ADDRESS LINE 1", address_line2: "NEW ADDRESS LINE 2", city: "NEW CITY", state: "AA", zipcode: 11111 ) { id name email password image description address_line1 address_line2 city state zipcode } }`
       });
-    user_mutation.body.data.userUpdate.should.have.property('image')
-    expect(user_mutation.body.data.userUpdate.image).to.eq('updated_img.png')
 
-    user_mutation.body.data.userUpdate.should.have.property('description')
-    expect(user_mutation.body.data.userUpdate.description).to.eq('new day, new me')
+    const mutated_user = user_mutation.body.data.userUpdate;
```
Comment (leahriffell) [inline code/api/src/modules/user/tests/userMutation.test.js:71]: mutated userrrr 🧟

File: code/api/src/modules/user/tests/userQuery.test.js [hunk 2]
```diff
@@ -114,39 +125,50 @@ describe('User queries', () => {
...
-    all_users_request.body.data.users[0].should.have.property('state')
-    expect(all_users_request.body.data.users[0].state).to.eq('CO')
+    admin.should.have.property('state')
+    expect(admin.state).to.eq('CO')
 
-    all_users_request.body.data.users[0].should.have.property('zipcode')
-    expect(all_users_request.body.data.users[0].zipcode).to.eq(36479);
+    admin.should.have.property('zipcode')
+    expect(admin.zipcode).to.eq(36479);
   });
```
Comment (leahriffell) [inline code/api/src/modules/user/tests/userQuery.test.js:173]: Thanks for adding back in all of this work you already did! @priyapower

File: code/api/src/modules/user/tests/userQuery.test.js [hunk 3]
```diff
@@ -196,7 +218,43 @@ describe('User queries', () => {
...
+  it('attempts query for genders typo in query function', async () => {
+    const bad_request = await request(server)
+      .post('/')
+      .send({ query: '{ userGender { id name } }' });
+    expect(bad_request.status).to.eq(400);
+    bad_request.body.should.have.property('errors')
+    bad_request.body.errors[0].should.have.property('message')
+    bad_request.body.errors[0].message.should.be.a('string');
+    expect(bad_request.body.errors[0].message).to.eq('Cannot query field "userGender" on type "query". Did you mean "userGenders"?');
+  });
```
Comment (leahriffell) [inline code/api/src/modules/user/tests/userQuery.test.js:257]: good sad path testing here

**Review by leahriffell — APPROVED** [review]

File: code/api/src/modules/user/tests/userMutation.test.js (hunk 1 above)
Comment (priyapower) [reply code/api/src/modules/user/tests/userMutation.test.js:71]: Ahhh!!! Scary user!

**Review by judithpillado — APPROVED** [review]

**Pull request merged**